Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using AzureCognitiveSearch.PowerSkills.Common;
+using Newtonsoft.Json.Linq;
+
+// languages used for Azure Search with Text Analytics:
+// el, th, he, tr, cs, hu, ar, ja-jp, fi, da, no, ko, pl, ru, sv, ja, it, pt, fr, es, nl, de, en
+// greek, thai, hebrew, turkish, czech, hungarian, arabic, japanese, finnish, danish, norwegian, korean, polish, russian, swedish, japanese (again??), 
+// italian, portuguese, french, spanish, dutch, german, english
+// unicode blocks in order:
+// InThai, InHebrew
+
+namespace AzureCognitiveSearch.PowerSkills.Text.CustomEntitySearch
+{
+    /// <summary>
+    /// Based on sample custom skill provided in Azure Search. Provided a user-defined list of entities
+    /// this function determines the Entity first occurrence within a given document. This list of entities
+    /// must repeatedly be provided by the user for each document.
+    /// </summary>
+    public static class CustomEntitySearch
+    {
+        private static int MaxRegexEvalTime = 1;
+        /// <summary>
+        /// We assert the following assumptions:
+        /// 1. All text files are Latin based (i.e. no languageCode)
+        /// 2. Words can contain special characters and numbers
+        /// 3. The provided entities are not case sensitive
+        /// </summary>
+
+        [FunctionName("custom-search")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log, ExecutionContext executionContext)
+        {
+            log.LogInformation("Custom Entity Search function: C# HTTP trigger function processed a request.");
+
+            string skillName = executionContext.FunctionName;
+            IEnumerable<WebApiRequestRecord> requestRecords = WebApiSkillHelpers.GetRequestRecords(req);
+            if (requestRecords == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
+            }
+
+            WebApiSkillResponse response = WebApiSkillHelpers.ProcessRequestRecords(skillName, requestRecords,
+                (inRecord, outRecord) => {
+                    string text = Regex.Escape(inRecord.Data["text"] as string);
+                    List<string> words = ((JArray)inRecord.Data["words"]).ToObject<List<string>>();
+                    if (words == null)
+                    {
+                        words = new WordLinker(executionContext.FunctionAppDirectory).Words;
+                    }
+
+                    var entities = new List<Entity>();
+                    var entitiesFound = new List<string>();
```

Comment:
I might have missed some context, but why are you returning two lists now?

Reply:
According to the emails, they want a list of all entities found as well as a more "detailed" version with the matchIndex and later the fuzzy correlation score

Reply:
So the issue here will be that you'll have duplicate items in `entitiesFound`. `entities` should be a list of matches (with index), wehreas `entitiesFound` should be a list of the found words with no duplicates. You could use a `HashSet` for the latter to easily get rid of the duplicate entries, and just `ToArray` it when you're done.

Reply:
this is still a list in iteration 10

---
In reply to: [305538620](https://github.com/Azure-Samples/azure-search-power-skills/pull/1#discussion_r305538620) [](ancestors = 305538620)